### DISCUSSION TITLE
feat: English - Turkish language support should be added to the application.

### DISCUSSION THREAD

**Discussion by ahmettulutas** [acikkaynak/deprem-yardim-frontend#597] [Ideas]
## Define your request
Language support should be added to the application with react-intl.

** discord username: @Ahmet Ulutaş#8174**

We can provide English - Turkish language support to the application by creating language files with react-intl.

https://vimeo.com/797375383/2b87da0f33

**Comment by sarpkayature** [comment]
@ahmettulutas i18n can be implemented. Here, we will need the necessary files for different languages.

**Comment by emreerdendev** [comment]
I removed the unchecked tag, but if you check the template according to it, it would be much better if you enter the discord username correctly. 
The link of the template: https://github.com/acikkaynak/deprem-yardim-frontend/blob/development/.github/ISSUE_TEMPLATE/feature_request.md

**Comment by 0xCakin** [comment]
@ahmettulutas, are you proposing to make the language of React in English? Can we get a bit more detail? 
@eraygundogmus, can you take a look here?"

**Comment by ahmettulutas** [comment]
> @ahmettulutas React'in dilini mi ingilizce olmasini teklif ediyorsun? Biraz daha detay alabilir miyiz? @eraygundogmus buraya bir bakabilir misin?

No. I can add a language feature to this application. That way, not only users who only speak Turkish, but also foreign users can use this application. It is even ready for development. If you approve, I can create a pull request. https://vimeo.com/797375383/2b87da0f33

**Comment by hasanugr** [comment]
@ahmettulutas It is important not to fill in the field on mobile currently. You have used a dropdown for Desktop, what have you worked on for mobile?

I have a few suggestions for this. Firstly, instead of selecting with a button, we can automatically give the device language by capturing it. The conditions I have created in my mind are as follows:
If the device language is TR or Unknown, the TR language can be shown.
In all cases except the above, the EN language can be displayed (as it is the global language).

**Comment by ahmettulutas** [comment]
Actually, this button works up to 280px width, so there is no issue of not being able to use it on mobile devices. Of course, the language can be given by capturing the device language, but if the device language is something other than English, the fallback will always be English. Otherwise, we need to create translation files for all device languages, in my humble opinion, there is no need for that. I added this for the default language to be Turkish, and for a foreign user to use it if he/she wishes, to have an English language option, and to be able to switch from Turkish to English, as it is practical and fast.

**Comment by hasanugr** [comment]
@ahmettulutas I understand what you said about the "no," but for the mobile side, it's important to narrow the area if it's over 280px to make it more user-friendly. That's why I suggested capturing the device language. Also, we don't need to make all the languages, if the device language is TR then show TR, for all other device languages show EN.

We can leave this to the product or the leads, my suggestion is just to keep the user interface simple.

**Comment by ahmettulutas** [comment]
> @ahmettulutas Yok dediğini anladım fakat mobile de 280px üstü ols abile alan daraltacak, kullanımı kolaylaştırmamız gerekiyor daha çok mobile için. Bu sebeple cihaz dilini yakalamayı önerdim, ayrıca tüm dilleri yapmamıza gerek yok, üstte de açıklamıştım TR ise TR, TR olmayan her cihaz dili için EN gösterilir.
> 
> Bunu product'a veya lead'lere bırakabiliriz, benimki kullanıcı arayüzünü basit tutmak için bir öneri sadece.

I found a conventional solution by using the next.js project's useRouter() hook and react-intl libraries. I can shrink the button to prevent it from taking up space on mobile.

**Comment by suyuti** [comment]
I think the application should support Turkish, English, and Arabic based on the demographic of the region.

**Reply by ahmettulutas** [reply]
Arapça desteğini de çok hızlı bir şekilde ekleyebiliriz. Şöyle bir json dosyası oluşturmam yeterli olur.
<img width="1125" alt="Screenshot 2023-02-09 at 21 32 26" src="https://user-images.githubusercontent.com/82840617/217905501-6fa79dc8-0bf2-4359-a44e-d2b8a296699e.png">
